### DISCUSSION TITLE
Best way to use SQLPage embedded in a Rust CLI?

### DISCUSSION THREAD

**Discussion by shah** [sqlpage/SQLPage#121] [Q&A] [answered]
Great package @lovasoa thanks for all the hard work. I am building a CLI for internal use cases and one part of the CLI helps manage an SQLite database. I'd like to be able to add a `server` command to my CLI which would launch SQLPage's web server bundled as a crate into my CLI. Do you have any hints or tips that might help with this? Thanks!

**Comment by lovasoa** [comment, marked as answer]
Hello ! 

  You may want to have a look at the tests for ideas of how to include sqlpage in an external crate: https://github.com/lovasoa/SQLpage/blob/main/tests/index.rs

  The root function that you'll want to use is probably: https://github.com/lovasoa/SQLpage/blob/main/src/webserver/http.rs#L562-L581
